database.go

```diff
@@ -0,0 +1,1071 @@
+	if err != nil {
+		return nil, nil, err
+	}
+
+	databaseDB := new(databaseDBBase)
+	resp, err := d.client.DoWithContext(ctx, req, databaseDB)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return databaseDB.DatabaseDB, resp, nil
+}
+
+// Get information on an individual logical database within a Managed Database based on a dbname and databaseID
+func (d *DatabaseServiceHandler) GetDB(ctx context.Context, databaseID string, dbname string) (*DatabaseDB, *http.Response, error) {
+	uri := fmt.Sprintf("%s/%s/dbs/%s", databasePath, databaseID, dbname)
+
+	req, err := d.client.NewRequest(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	databaseDB := new(databaseDBBase)
+	resp, err := d.client.DoWithContext(ctx, req, databaseDB)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return databaseDB.DatabaseDB, resp, nil
+}
+
+// Delete a user within the Managed database. All data will be permanently lost.
+func (d *DatabaseServiceHandler) DeleteDB(ctx context.Context, databaseID string, dbname string) error {
+	uri := fmt.Sprintf("%s/%s/dbs/%s", databasePath, databaseID, dbname)
+
+	req, err := d.client.NewRequest(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return err
+	}
+
+	_, err = d.client.DoWithContext(ctx, req, nil)
+	return err
+}
+
+// List all available maintenance updates for your Managed Database.
+func (d *DatabaseServiceHandler) ListMaintenanceUpdates(ctx context.Context, databaseID string) ([]string, *http.Response, error) {
+	uri := fmt.Sprintf("%s/%s/maintenance", databasePath, databaseID)
+
+	req, err := d.client.NewRequest(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	databaseUpdates := new(databaseUpdatesBase)
+	resp, err := d.client.DoWithContext(ctx, req, databaseUpdates)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return databaseUpdates.AvailableUpdates, resp, nil
+}
+
+// Start the maintenance update process for your Managed Database
+func (d *DatabaseServiceHandler) StartMaintenance(ctx context.Context, databaseID string) (string, *http.Response, error) {
+	uri := fmt.Sprintf("%s/%s/maintenance", databasePath, databaseID)
+
+	req, err := d.client.NewRequest(ctx, http.MethodPost, uri, nil)
+	if err != nil {
+		return "", nil, err
+	}
+
+	databaseUpdates := new(databaseMessage)
+	resp, err := d.client.DoWithContext(ctx, req, databaseUpdates)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return databaseUpdates.Message, resp, nil
+}
+
+// Query for service alerts for the Managed Database using the given parameters
+func (d *DatabaseServiceHandler) ListServiceAlerts(ctx context.Context, databaseID string, databaseAlertsReq *DatabaseListAlertsReq) ([]DatabaseAlert, *http.Response, error) {
+	uri := fmt.Sprintf("%s/%s/alerts", databasePath, databaseID)
+
+	req, err := d.client.NewRequest(ctx, http.MethodPost, uri, databaseAlertsReq)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	databaseAlerts := new(databaseAlertsBase)
```

Comment:
Dont do this, use a `NewX` pattern instead, see: https://github.com/tmrts/go-patterns/blob/master/creational/factory.md

Reply:
The entire govultr library does this for all of the API endpoints through the way it interacts with the DoWithContext function, so we probably want to refactor that down the road since it would all have to be done together for the whole codebase.